ci/check-protobuf-stability.sh

```diff
@@ -89,26 +89,33 @@ USAGE
   exit
   ;;
 --stable)
-  if [[ $TARGET == "main" ]]; then
-    # Check against the most recent stable tag.
-    # This check runs only on main or PRs targeting main.
-    #
-    # This check does not need to run on release branch commits because
-    # they are built sequentially, so no conflicts are possible and the per-PR
-    # check is enough.
-    readonly LATEST_STABLE_TAG="$(git tag | grep -v "snapshot" | sort -V | tail -1)"
-    # The v1.17 stable release includes the buf config file with the default name `buf.yml`.
-    # Starting with v1.18 we have multiple buf config files.
-    if [[ $LATEST_STABLE_TAG =~ "v1.17."* ]]; then
-      BUF_CONFIG_UPDATED=false
-    else
-      BUF_CONFIG_UPDATED=true
-    fi
-    BUF_GIT_TARGET_TO_CHECK=".git#tag=${LATEST_STABLE_TAG}"
-    check_protos
+  # Check against the most recent stable tag.
+  #
+  # This check does not need to run on release branch commits because
+  # they are built sequentially, so no conflicts are possible and the per-PR
+  # check is enough.
+  readonly RELEASE_BRANCH_REGEX="^release/.*"
+  LATEST_STABLE_TAG=""
+  if [[ "${TARGET}" =~ ${RELEASE_BRANCH_REGEX} ]]; then
+    readonly VERSION="${TARGET#release/}"
+    readonly VERSION_PREFIX="${VERSION%.x}"
+    readonly STABLE_TAGS=($(git tag | grep "v.*" | grep -v "snapshot" | sort -V))
+    LATEST_STABLE_TAG="$(for TAG in "${STABLE_TAGS[@]}"; do if [[ $(semver compare "${TAG#v}" "${VERSION_PREFIX}.999") == "-1" ]]; then echo "$TAG"; fi; done | tail -1)"
+  elif [[ "${TARGET}" == "main" ]]; then
+    LATEST_STABLE_TAG="$(git tag | grep "v.*" | grep -v "snapshot" | sort -V | tail -1)"
+  else
+    echo "unsupported target branch $TARGET" >&2
+    exit 1
+  fi
+  # The v1.17 stable release includes the buf config file with the default name `buf.yml`.
+  # Starting with v1.18 we have multiple buf config files.
+  if [[ $LATEST_STABLE_TAG =~ "v1.17."* ]]; then
```

Comment:
Would this be cleaner as a `semver compare` call? What happens if we somehow need to make a patch release to 1.16?

Reply:
@nicu-da, @fabiotudone-da, do you know what was the idea behind that. Shall we change it to `semver compare` to make it work with 1.16 and older?

Reply:
[This](https://github.com/digital-asset/daml/pull/12246/files#diff-02e1599226c88e08201722fd81f5919a2f35bf9659a87af517abba07c12fa82aR47) is the relevant comment. @garyverhaegen-da Why would a 1.16 patch be a problem (unless we separate the buf files in that patch, which I don't expect)? I agree it would be cleaner to always use `semver`, though.

Reply:
I don't have much context here. What I see is a comment that says "things are different after 1.17" and a code that checks "== 1.17", whereas based on the comment I'd have expected the code to check "<= 1.17". If you think the code is right please clarify the comment.

Reply:
I think you're actually right, for some reason I thought it was a less-than-or-equal comparison.

Reply:
Thanks Gents, now it works for <= 1.17.